### PR TITLE
feat: eBird-compatible sightings export and roundtrip import

### LINKED ISSUE #73: Export should conform to what eBird can import
Right now just raw data that can't even be imported back to BirdDex

### PR DESCRIPTION
**Pull request by jlian**
## Summary

Closes #73. Makes the sightings CSV export conform to eBird's "My eBird Data" format so it can be re-imported back into WingDex (and theoretically into eBird).

### Export roundtrip

- Sightings CSV now includes all eBird-required columns (Submission ID, Common Name, Taxonomic Order, Count, State/Province, Country Code, Protocol, Duration, etc.)
- Export populates outing-level metadata: region, protocol, observer count, effort fields
- Re-importing an exported CSV produces identical observations (duplicate detection by date+species)

### Schema (migration 0005)

- **outing**: added stateProvince, countryCode, protocol, numberObservers, allObsReported, effortDistanceMiles, effortAreaAcres
- **observation**: added speciesComments

Migration 0005 is applied automatically by the release workflow before deploy (both preview and production).

### Bug fixes

- Region fields now persist correctly on partially-migrated DBs (decoupled column capability checks)
- Photo POST race condition resolved (retry with delay when outing has not persisted yet)
- Observation ID collisions eliminated (crypto.randomUUID instead of Date.now)
- Lifer confetti fires per-outing instead of only on the final outing
- Upload toast durations tuned (lifer 3s, saved 8s)

### Tests

- Realistic variant eBird CSV fixture coverage
- Duplicate re-import and CSV schema edge-case tests
- UI import flow coverage for variant CSVs
- Parity enforcement between demo and e2e fixtures

### CODE REVIEW

File: functions/api/import/ebird-csv/confirm.ts [hunk 1]
```diff
@@ -53,48 +54,128 @@ export const onRequestPost: PagesFunction<Env> = async context => {
...
+          )
+          .bind(
+            observation.id,
+            observation.outingId,
+            userId,
+            observation.speciesName,
+            observation.count,
+            observation.certainty,
+            observation.notes || null,
+            null
```
[the remark this answers is not in the document]
Comment (jlian) [reply functions/api/import/ebird-csv/confirm.ts]: Fixed in d5e3466. In the speciesComments-supported insert path, notes is now bound to an empty string to satisfy the NOT NULL constraint on observation.notes.
